### ISSUE TITLE
Trailing slash impacts match

### ISSUE THREAD

**Issue by Y0hy0h** [micromatch/micromatch#128]
## Please describe the **minimum necessary steps** to reproduce this issue:
```js
var micromatch = require("micromatch")
console.log(micromatch(['t'], ['t/']));  => []
console.log(micromatch(['t'], ['t']));   => ["t"]
```

## What is happening (but shouldn't):
The trailing slash prevents the pattern from matching.

## What should be happening instead?
I haven't figured out how to test this behaviour against bash, but I assume that the trailing slash should still match.

This behaviour came up in [an issue on yarn](https://github.com/yarnpkg/yarn/issues/5816). There, the desired behaviour is for both patterns to match the file.

**Comment by Y0hy0h** [comment]
Sorry, could have checked the version. 😅 

Thanks for the lightning fast reply. :)

**Comment by Y0hy0h** [comment]
Actually, in the newest version, the bug is present. See https://runkit.com/embed/h73y8gc3id4a.

**Comment by jonschlinkert** [comment]
> bug

Is this an expected result in bash?

**Comment by Y0hy0h** [comment]
Ok, I've finally found a link on how to test this in bash. (It's really easy. 🤦‍♂️ )

> Is this an expected result in bash?

I can't tell. If at my root folder I run `echo bin`, it returns `bin`, if I run `echo bin/`, it returns `bin/`. To me, this is ambiguous with respect to this issue.

I just interpreted [your response](https://github.com/micromatch/micromatch/issues/128#issuecomment-388967459) as agreeing with this issue. But ultimately, you decide if this is expected for micromatch. :)

So is this behaviour intended in micromatch, or is it considered a bug?

**Comment by jonschlinkert** [comment]
> To me, this is ambiguous with respect to this issue.

I agree, this issue isn't as clear-cut as it seems at first glance. In globbing libs that hit the fs - as with bash, it's easier to reason that a trailing slash should match directories, but there is no concept of a directory here. On one hand, I agree that (intuitively) it just seems like it should match. But given that we don't have the benefit of file system hints, et cetera, I'm more inclined to say that it should be more of an "exact match", which is the [current (intentional) behavior](https://github.com/micromatch/micromatch/blob/master/test/globstars.js#L77). 

I'm leaving this open for discussion though.

**Comment by Y0hy0h** [comment]
Yes, this is absolutely fine. :)

Thanks for your response!
